RockLib.Logging/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -217,7 +218,7 @@ private static void SetLoggerLookupDescriptor(this IServiceCollection services,
             // Immediately dispose any non-singleton loggers that weren't selected.
             for (var i = 0; i < loggers.Length; i++)
             {
-                if (!isSingletonLoggers[i] && !ReferenceEquals(loggers[i], selectedLogger))
+                if (!singletonLoggers.Any(s => ReferenceEquals(s, loggers[i])) && !ReferenceEquals(loggers[i], selectedLogger))
```

Comment:
This....should work. I think. The only way it wouldn't is if `AddLogger()` was called again to register another logger, which....frankly, you really shouldn't do anyway.

Reply:
Tests will help ;)